### ISSUE TITLE
Ligatures not supported

### ISSUE THREAD

**Issue by Kronuz** [babel/babel-sublime#343]
In Sublime Text 3 build 3146 (October 6, 2017) they added ligature support for symbols; it works in their JavaScript syntax, but not in JavaScript (Babel). That means the Babel syntax isn't properly scoping some symbols/operators.

It's needed to "correct" scoping of operators. For instance, `=>` should be a single token, not `=` with one set of scopes and `>` with another set.

**Comment by Kronuz** [comment]
Actually ligatures are supported, however... in function arrows that have parameters in multiple lines they're not:

```javascript
// the function arrow ligature works as expected in this snippet
const goodFunctionArrow = (param) => {}; // param scopes: variable.parameter.function.js
```

```javascript
const badFunctionArrow = (
  // an extra line here makes the function arrow below have different
  // scopes for each of its symbols, thus removing the ligature:
  // keyword.operator.assignment.js and keyword.operator.relational.js
  param // param scopes: variable.other.readwrite.js
) => {};
```

It might be related to the fact the first example shows `param` with different scopes.

**Comment by borela** [comment]
This is a limitation in sublime's regex engine, it can only see one line at a time, in my [syntax](https://github.com/borela/naomi/issues/25), I added a fallback where if it sees a `=>` it will still highlight the function body but the parameters will not have the correct scopes.

While this is not ideal, it helps while support for multi line regexes is not implemented https://github.com/SublimeTextIssues/Core/issues/1693

**Comment by sunny-g** [comment]
I'm having the same issue with the pipeline `|>` operator, regardless of the number of lines the function is written.

**Comment by borela** [comment]
@sunny-g That's because this syntax does not support it, try https://github.com/borela/naomi

**Comment by Thom1729** [comment]
This should be fixed in v10.0.0-beta.1. See #379 for more information on the new release. Please report any problems with v10 in a new issue.
